packages/ra-core/src/controller/create/useCreateController.spec.tsx

```diff
@@ -502,4 +503,34 @@ describe('useCreateController', () => {
             expect.any(Function)
         );
     });
+
+    it('The save function should return errors from the create call', async () => {
+        const create = jest.fn().mockImplementationOnce(() => {
+            return Promise.reject({ body: { errors: { foo: 'invalid' } } });
+        });
+        const dataProvider = ({
+            create,
+        } as unknown) as DataProvider;
+        let saveCallback;
+        render(
+            <CoreAdminContext dataProvider={dataProvider}>
+                <CreateController {...defaultProps}>
+                    {({ save, record }) => {
+                        saveCallback = save;
+                        return <div />;
+                    }}
+                </CreateController>
+            </CoreAdminContext>
+        );
+        await new Promise(resolve => setTimeout(resolve, 10));
+        let errors;
+        await act(async () => {
+            errors = await saveCallback({ foo: 'bar' });
+        });
+        expect(errors).toEqual({ foo: 'invalid' });
+        await new Promise(resolve => setTimeout(resolve, 10));
```

Comment:
I don't understand why you need to wait here. If you have errors, two lines above, it's sure that the dataProvider was already called, right?